packages/perennial-oracle/contracts/ChainlinkFeedOracle.sol

```diff
@@ -34,22 +35,43 @@ contract ChainlinkFeedOracle is IOracleProvider {
     /**
      * @notice Initializes the contract state
      * @param aggregator_ Chainlink price feed aggregator
+     * @param phases_ Array of phases to initialize the oracle with
+     * @dev If `phases_` is empty, the oracle will be initialized with the latest round from the aggregator as the
+     *      starting round
      */
-    constructor(ChainlinkAggregator aggregator_) {
+    constructor(ChainlinkAggregator aggregator_, Phase[] memory phases_) {
         aggregator = aggregator_;
 
         _decimalOffset = SafeCast.toInt256(10 ** aggregator.decimals());
 
-        ChainlinkRound memory firstSeenRound = aggregator.getLatestRound();
+        if (phases_.length > 0) {
+            // Phases should be initialized with at least 2 values
+            if (phases_.length < 2) revert InvalidPhaseInitialization();
 
-        // Load the phases array with empty phase values. these phases will be invalid if requested
-        while (firstSeenRound.phaseId() > _phases.length) {
-            _phases.push(Phase(0, 0));
-        }
+            // Phases[0] should always be empty, since phases are 1-indexed
+            if (phases_[0].startingVersion != 0 || phases_[0].startingRoundId != 0) revert InvalidPhaseInitialization();
+
+            // Phases[1] should start at version 0
+            if (phases_[1].startingVersion != 0) revert InvalidPhaseInitialization();
+
+            // Load phases array with the provided phases
+            for (uint i = 0; i < phases_.length; i++) {
+                _phases.push(phases_[i]);
+            }
+            // Set the lastSyncedRoundId to the starting round of the latest phase
+            _lastSyncedRoundId = phases_[phases_.length - 1].startingRoundId;
```

Comment:
Because this "resets" the `_lastSyncedRoundId ` when we deploy a new implementation, there could be a case where a phase shift occurs between construction and the first sync() call, in which case we'd be out of sync from the original instance. I think we just need to call sync()? or maybe we need to be smarter about how we set _lastSyncedRound?

Reply:
Yeah I like 1 the best here -- I think either way there might be some slight mismatch if a phase shift occurs while we're in the process of switching over, but this should minimize that risk.